input/elasticapm/processor.go

```diff
@@ -296,19 +296,25 @@ func (p *Processor) handleStream(
 	if n == 0 {
 		return readErr
 	}
-
-	if err := p.processBatch(ctx, processor, batch); err != nil {
+	if err := processor.ProcessBatch(ctx, batch); err != nil {
 		return fmt.Errorf("cannot process batch: %w", err)
 	}
+	for _, v := range *batch {
```

Comment:
Can you add some unit tests where these counters are verified?

Reply:
Extended test `TestHandleStream` to check proper population.